pkg/controller/instanceset/in_place_update_util.go

```diff
@@ -157,6 +161,10 @@ func mergeInPlaceFields(src, dst *corev1.Pod) {
 			}
 		}
 	}
+	// remove role label if images are updated
+	if isImageUpdated {
+		delete(dst.Labels, constant.RoleLabelKey)
```

Comment:
Is it possible that` kb-agen`t will label the pod again?

Reply:
It's indeed possible. But the possibility is very low and the role label will be correct finally.